packages/ai/src/providers/google/index.ts

```diff
@@ -66,28 +66,41 @@ const createChatCompletion: ICreateChatCompletion = async (body) => {
   const { model, messages } = body;
   const googleAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY!);
   const timeout = body.timeout || DEFAULT_TIMEOUT;
-  const modelInstance = googleAI.getGenerativeModel(
-    { model },
-    { apiVersion: "v1beta", timeout },
-  );
+  const isBetaModel = model.includes("gemini-1.5");
+  // Google's JS library does not fully support Gemini 1.5 Pro
+  // We have an open issue with details:
+  // https://github.com/google/generative-ai-js/issues/98
+  const modelInstance = isBetaModel
+    ? googleAI.getGenerativeModel({
+        model: "gemini-pro",
+      })
+    : googleAI.getGenerativeModel({ model }, { apiVersion: "v1beta", timeout });
   const contents = massageOpenAIMessagesToGoogleAI(messages);
   const { executionDone } = await batch.waitForTurn();
   try {
     const startedAt = Date.now();
     const completion = await promiseRetry<GenerateContentResult>(
       (retry) => {
-        // TODO: move to model.startChat which support model config (e.g. temperature)
-        return modelInstance
-          .generateContent({ contents })
-          .catch((err: Error) => {
-            // TODO: Replace with instanceof checks when the Gemini SDK exports errors
-            console.log(err.message);
-            if (err.message.includes("[429 Too Many Requests]")) {
-              console.log("Attempting to retry");
-              retry(err);
-            }
-            throw err;
-          });
+        return (
+          isBetaModel
+            ? modelInstance.generateContent({ contents })
```

Comment:
should we do a spread on model parameters for the generateContent call as well? 

given the code is a bit hairy, this could use a test that just ensures we never regress on scenarios. setting up the test is easy (since we have a few examples now). but will need additional config on the CI to set the environment variable for GOOGLE_API_KEY. will leave it up to you whether you want to add the test - not a blocker

Reply:
Would be better to have the test. Will update this today. Missed the notifications earlier

Reply:
cool, call me if you need help in setting up test + CI

Reply:
Added the test, will need help adding the API key